addons/link_tracker/models/mail_render_mixin.py

```diff
@@ -40,19 +40,20 @@ def _shorten_links(self, html, link_tracker_vals, blacklist=None, base_url=None)
             long_url = match[1]
             label = (match[3] or '').strip()
 
-            if not blacklist or not [s for s in blacklist if s in long_url] and not long_url.startswith(short_schema):
+            if not (long_url.startswith(short_schema) or (blacklist and [s for s in blacklist if s in long_url])):
                 create_vals = dict(link_tracker_vals, url=utils.unescape(long_url), label=utils.unescape(label))
                 link = self.env['link.tracker'].search_or_create(create_vals)
                 if link.short_url:
-                    new_href = href.replace(long_url, link.short_url)
+                    # Avoiding Markup's overhead required to replace "&" characters, common in urls
+                    new_href = markupsafe.Markup(str.replace(href, long_url, link.short_url))
```

Comment:
Hello, members of https://github.com/orgs/odoo/teams/rd-security , could you please review this usage?